packages/manager/src/views/WebPropertyDetailPage/components/SSR/SSRDetails.tsx

```diff
@@ -0,0 +1,206 @@
+    }
+    setIsRedployModalOpen((prevIsModalOpen) => !prevIsModalOpen);
+  };
+
+  const handleConfigureModal = () => {
+    setIsConfigureModalOpen((prevIsModalOpen) => !prevIsModalOpen);
+  };
+
+  const url = window.location.href;
+  const parts = url.split('/');
+  const applicationName = parts[parts.length - 1];
+  const temp = spaProperties?.data?.[applicationName].filter((item) => item.isSSR === true);
+
+  return (
+    <>
+      {spaProperties.isSuccess && isSpaPropertyListEmpty ? (
+        <EmptyInfo propertyIdentifier={propertyIdentifier} />
+      ) : (
+        <TableComposable aria-label="spa-property-list" className="">
+          <>
+            <Caption>SSR&apos;s DEPLOYED</Caption>
+            <Thead noWrap>
+              <Tr>
+                <Th textCenter>SPA Name</Th>
+                <Th textCenter>Environments</Th>
+                <Th textCenter>Ref</Th>
+                <Th textCenter>Path</Th>
+                <Th textCenter>HealthCheck Path</Th>
+                <Th textCenter style={{ justifyContent: 'space-evenly', display: 'grid' }}>
+                  Actions
+                </Th>
+              </Tr>
+            </Thead>
+          </>
+          {(spaProperties.isLoading && webProperties.isLoading) ||
+          (spaProperties.isLoading && isSpaPropertyListEmpty) ? (
+            <TableRowSkeleton rows={3} columns={4} />
+          ) : (
+            <Tbody>
+              {temp?.map((val) => (
+                <Tr key={val.name}>
+                  <Td textCenter>{val?.name}</Td>
+                  <Td textCenter>
+                    <Label
+                      key={val.env}
+                      color={val.isSSR ? 'blue' : 'gold'}
+                      isCompact
+                      style={{ marginRight: '8px' }}
+                    >
+                      {val.env}
+                      {val.isSSR && '[ssr]'}
+                    </Label>
+                  </Td>
+                  <Td textCenter>{val?.ref}</Td>
+                  <Td textCenter>{val?.path}</Td>
+                  <Td textCenter>{val?.healthCheckPath}</Td>
+                  <Td textCenter style={{ justifyContent: 'flex-end', display: 'grid' }}>
+                    <SplitItem isFilled>
+                      <Button
+                        variant="primary"
+                        isSmall
+                        icon={<PencilAltIcon />}
+                        onClick={() => {
+                          handleConfigureModal();
+                          setConfigureData(val);
+                        }}
+                      >
+                        Configure
+                      </Button>
+                      &nbsp;&nbsp;
+                      <Button
+                        variant="secondary"
+                        isSmall
+                        icon={<UndoIcon />}
+                        onClick={() => {
+                          handleRedeployModal();
+                          setRedeployData(val);
```

Comment:
You could have used `usePopUp` hook used in the app for these kinda operations It handles intermediate data passing and lot more